pkg/reconciler/route/domains/domains_test.go

```diff
@@ -0,0 +1,183 @@
+				},
+			},
+		},
+		Network: &network.Config{
+			DefaultClusterIngressClass: "ingress-class-foo",
+			DomainTemplate:             network.DefaultDomainTemplate,
+		},
+		GC: &gc.Config{
+			StaleRevisionLastpinnedDebounce: time.Duration(1 * time.Minute),
+		},
+	}
+}
+
+func TestDomainNameFromTemplate(t *testing.T) {
+	type args struct {
+		name string
+	}
+	tests := []struct {
+		name     string
+		template string
+		args     args
+		want     string
+		wantErr  bool
+	}{{
+		name:     "Default",
+		template: "{{.Name}}.{{.Namespace}}.{{.Domain}}",
+		args:     args{name: "test-name"},
+		want:     "test-name.default.example.com",
+	}, {
+		name:     "Dash",
+		template: "{{.Name}}-{{.Namespace}}.{{.Domain}}",
+		args:     args{name: "test-name"},
+		want:     "test-name-default.example.com",
+	}, {
+		name:     "Short",
+		template: "{{.Name}}.{{.Domain}}",
+		args:     args{name: "test-name"},
+		want:     "test-name.example.com",
+	}, {
+		name:     "SuperShort",
+		template: "{{.Name}}",
+		args:     args{name: "test-name"},
+		want:     "test-name",
+	}, {
+		// This cannot get through our validation, but verify we handle errors.
+		name:     "BadVarName",
+		template: "{{.Name}}.{{.NNNamespace}}.{{.Domain}}",
+		args:     args{name: "test-name"},
+		wantErr:  true,
+	}}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			route := &v1alpha1.Route{
+				ObjectMeta: metav1.ObjectMeta{
+					SelfLink:  "/apis/serving/v1alpha1/namespaces/test/Routes/myapp",
+					Name:      "myroute",
+					Namespace: "default",
+					Labels: map[string]string{
+						"route": "myapp",
+					},
+				},
+			}
+
+			ctx := context.Background()
+			cfg := testConfig()
+			cfg.Network.DomainTemplate = tt.template
+			ctx = config.ToContext(ctx, cfg)
+
+			got, err := DomainNameFromTemplate(ctx, route, tt.args.name)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("BuildSubRouteWithName() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if got != tt.want {
+				t.Errorf("BuildSubRouteWithName() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestSubdomainName(t *testing.T) {
+	route := &v1alpha1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			SelfLink:  "/apis/serving/v1alpha1/namespaces/test/Routes/myapp",
+			Name:      "myroute",
+			Namespace: "default",
+			Labels: map[string]string{
+				"route": "myapp",
+			},
+		},
+	}
+
+	type args struct {
+		suffix string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "has suffix",
+			args: args{suffix: "mysuffix"},
+			want: "myroute-mysuffix",
+		}, {
+			name: "no suffix",
+			args: args{},
+			want: "myroute",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := SubdomainName(route, tt.args.suffix); got != tt.want {
+				t.Errorf("BuildName() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestTagURL(t *testing.T) {
+	type args struct {
```

Comment:
it's more useful here, but I am still unsure that we need it. Until the number of params gets unwieldy I think we should just list the args in the test itself.

Reply:
These were auto generated, so I just rolled with it. No problem, will squash it!